src/worker/api/index.ts

```diff
@@ -1437,8 +1486,17 @@ const getTradingBlockOffers = async (pids: number[], dpids: number[]) => {
 		);
 	};
 
-	const offers = await getOffers(pids, dpids);
-	return augmentOffers(offers);
+	//since filters reduce the number of viable offers, try up to 10 times to get some good offers
+	let filteredOffers: OfferType[] = [];
+	let iterations = 0;
+	while (filteredOffers.length === 0 && iterations < 10) {
```

Comment:
Is it possible for this to result in your getting the same (or very similar) offers twice? I don't see a team restriction anywhere.

Reply:
Are you referring to subsequent offer requests? Can you elaborate? The filters are applied as the final step, so that aspect shouldn't be any different from how offers initially worked

Reply:
Previously `getOffers` was called once. Now it's called up to 10 times. But each time it's called, some of the offers returned could be duplicates of previous ones, right?

Reply:
So if it gets *any* offers, it will return them (even if just 1) and display them to the player. Definitely possible/likely that it asks a team multiple times-- but sometimes the offer will be slightly different and may end up meeting the conditions that a previous one didn't